graphql_compiler/compiler/emit_match.py

```diff
@@ -127,18 +155,32 @@ def emit_code_from_ir(match_query):
         raise AssertionError(u'Unexpected falsy value for match_query.match_traversals received: '
                              u'{} {}'.format(match_query.match_traversals, match_query))
 
+    # Represent and add the MATCH traversal steps.
     match_traversal_data = [
         _represent_match_traversal(x)
         for x in match_query.match_traversals
     ]
-
     query_data.append(match_traversal_data[0])
     for traversal_data in match_traversal_data[1:]:
         query_data.append(u', ')
         query_data.append(traversal_data)
 
     query_data.appendleft(u' (')            # Prepare to wrap the MATCH in a SELECT.
     query_data.append(u'RETURN $matches)')  # Finish the MATCH query and the wrapping ().
+
+    # Represent and add the LET clauses for any @fold scopes that might be part of the query.
+    fold_data = [
+        _represent_fold(fold_location, fold_ir_blocks)
+        for fold_location, fold_ir_blocks in six.iteritems(match_query.folds)
+    ]
+    if fold_data:
+        query_data.append(u' LET ')
+        query_data.append(fold_data[0])
+        for fold_clause in fold_data[1:]:
+            query_data.append(u', ')
```

Comment:
Wait - Why are we appending this to every string (except for the first one) in the folded blocks ?

Reply:
Because the different clauses in the `LET` section have to be comma-separated. We append the first clause, and then we append a comma before every subsequent clause.